tests/integration/apply_ip_interface_test.py

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+import pytest
+
+from libnmstate import netapplier
+
+
+@pytest.fixture
+def eth1_state():
+    return {
+        'interfaces': [
+            {
+                'name': 'eth1',
+                'state': 'up',
+                'type': 'ethernet',
+                'mtu': 1500,
+                'ipv4': {
+                    'enabled': False
+                },
+                'ipv6': {
+                    'enabled': False
+                }
+            }
+        ]
+    }
+
+
+def test_add_static_ipv4(eth1_state):
+    eth1_state['interfaces'][0]['ipv4']['enabled'] = True
+    eth1_state['interfaces'][0]['ipv4']['address'] = [
+        {'ip': '192.168.122.250', 'prefix-length': 24}
+    ]
+    netapplier.apply(eth1_state)
```

Comment:
IMHO some kind of assertion should be added after applying the state.

Reply:
@tyll , I agree. I am waiting for the dependencies to get merged to make it easier for me to work on this.